course7/homeworks/mathias/complex_rpn/main.cpp

```diff
@@ -0,0 +1,30 @@
+#include "rpn_calculator.h"
+#include "complex.h"
+
+#include <iostream>
+#include <string>
+
+int main(int , char** )
+{
+	RpnCalculator calc;
+	std::string line;
+
+	std::cout << "> ";
+	while (std::getline(std::cin, line))
+	{
+		try
+		{
+			calc.parse(line);
+		}
+		catch (const std::runtime_error& e)
+		{
+			std::cout << "Error: " << e.what() << std::endl;
```

Comment:
1. If the error is not recoverable, then let the exception flow out and kill the process. In this case, any error makes the state inconsistent, as any subsequent commands can make assumptions/expectations regarding the state of the stack.
2. If errors are recoverable, hen print them to STDERR (```std::cerr```) to not pollute the output. When writing a Unix tool, always expect that someone may want to parse your output.